sandbox/cgroup/scanning.go

```diff
@@ -0,0 +1,155 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package cgroup
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/snapcore/snapd/snap/naming"
+)
+
+var (
+	// string that looks like an hook security tag
+	taggish1 = regexp.MustCompile(`snap\.[^.]+\.hook\.[^.]+`)
+	// string that looks like an app security tag
+	taggish2 = regexp.MustCompile(`snap\.[^.]+\.[^.]+`)
+)
+
+// securityTagFromCgroupPath returns a security tag from cgroup path.
+func securityTagFromCgroupPath(path string) (securityTag string) {
+	leaf := filepath.Base(filepath.Clean(path))
+
+	// We are only interested cgroup directory names that correspond to
+	// services and scopes, as they contain processes that have been invoked
+	// from a snap.
+	if ext := filepath.Ext(leaf); ext != ".service" && ext != ".scope" {
+		return ""
+	}
+
+	// There are two broad forms expressed by taggish1 and taggish2 regular expressions.
+	for _, re := range []*regexp.Regexp{taggish1, taggish2} {
+		if maybeTag := re.FindString(leaf); maybeTag != "" {
+			// NOTE: The things we are returning need to minimally look like
+			// security tags. They are further refined and looked at below, in
+			// PidsOfSnap.
```

Comment:
this note is a bit confusing to me, given that we check they are valid at least, and also below we mostly check that the prefix is for the snap we want, so they are quite good already, so not sure what is meant here with minimally?

Reply:
I think this comment is stale and pre-dates tag validation that happened since. I will revisit all the comments in the patch to ensure there's nothing like this.